lib/api_repo.js

```diff
@@ -1018,11 +1021,14 @@ ApiRepo.prototype._makeProtocFunc = function _makeProtocFunc(opts, language) {
           pluginOption = '--plugin=';
         } else {
           args.push('--' + language + '_out=' + outDir);
-          if (language === 'python') {
-            // required per https://github.com/grpc/grpc/issues/7857
-            args.push('--grpc_python_out=' + outDir);
-          } else {
-            args.push('--grpc_out=' + outDir);
+          if (!opts.buildCommonProtos ||
+              _.includes(COMMON_GRPC_PROTOS, protoPath)) {
+            if (language === 'python') {
+              // required per https://github.com/grpc/grpc/issues/7857
+              args.push('--grpc_python_out=' + outDir);
```

Comment:
After offline discussion, I understand that this is actually an improvement over the current state of master, where every common-proto is being built with the gRPC plugin (and hence potentially GAE-incompatible).